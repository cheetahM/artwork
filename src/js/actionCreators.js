// @flow

import axios from "axios";
import { SET_SEARCH_MOVIE, ADD_API_DATA } from "./actions";

export function setSearchMovie(searchMovie: string) {
  return { type: SET_SEARCH_MOVIE, payload: searchMovie };
}

export function addAPIData(apiData: Movie) {
  return { type: ADD_API_DATA, payload: apiData };
}

export function getAPIDetails(movieId: string) {
  return (dispatch: Function) => {
    axios
      .get(`http://localhost:3000/details/${movieId}`)
      .then(response => {
        dispatch(addAPIData(response.data));
      })
      .catch(error => {
        console.log("API error: ", error); // eslint-disable-line no-console
      });
  };
}
