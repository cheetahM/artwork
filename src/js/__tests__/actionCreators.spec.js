// @flow

import moxios from "moxios";
import { setSearchMovie, addAPIData, getAPIDetails } from "../actionCreators";

const orangeIsNewBlack = {
    movieId: 70242311,
    movieName: "Orange Is the New Black",
    imageType: "sdp",
    thumbnailUrl: "http://art.nflximg.net/304d2/7da9da4ea90c6df7889b67137cb64737e65304d2.jpg",
    fullSizeImageUrl: "http://art.nflximg.net/934f7/e76e738da86e04d9c388605789211d6a883934f7.jpg",
    languageCode: "de"
};

test("setSearchMovie", () => {
    expect(setSearchMovie("en")).toMatchSnapshot();
});

test("addAPIData", () => {
    expect(addAPIData(orangeIsNewBlack)).toMatchSnapshot();
});

test("getAPIDetails", (done: Function) => {
    const dispatchMock = jest.fn();
    moxios.withMock(() => {
        getAPIDetails(orangeIsNewBlack.movieId)(dispatchMock);
        moxios.wait(() => {
            const request = moxios.requests.mostRecent();
            request
                .respondWith({
                    status: 200,
                    response: orangeIsNewBlack
                })
                .then(() => {
                    expect(request.url).toEqual(`http://localhost:3000/details/${orangeIsNewBlack.movieId}`);
                    expect(dispatchMock).toBeCalledWith(addAPIData(orangeIsNewBlack));
                    done();
                })
        })
    })
});