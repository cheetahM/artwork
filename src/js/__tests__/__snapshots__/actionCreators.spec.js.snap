// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "fullSizeImageUrl": "http://art.nflximg.net/934f7/e76e738da86e04d9c388605789211d6a883934f7.jpg",
    "imageType": "sdp",
    "languageCode": "de",
    "movieId": 70242311,
    "movieName": "Orange Is the New Black",
    "thumbnailUrl": "http://art.nflximg.net/304d2/7da9da4ea90c6df7889b67137cb64737e65304d2.jpg",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchMovie 1`] = `
Object {
  "payload": "en",
  "type": "SET_SEARCH_MOVIE",
}
`;
